src/libstd/Cargo.toml

```diff
@@ -15,6 +15,7 @@ crate-type = ["dylib", "rlib"]
 
 [dependencies]
 alloc = { path = "../liballoc" }
+cfg-if = "0.1.8"
```

Comment:
Shouldn't this enable the `rustc-dep-of-std` feature?

Reply:
Hm, that leads to a build failure?!?
```
error: failed to select a version for `cfg-if`.
    ... required by package `std v0.0.0 (/home/r/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd)`
    ... which is depended on by `sysroot v0.0.0 (/tmp/xargo.NWDhLbN6RoeP)`
versions that meet the requirements `^0.1.9` are: 0.1.9

the package `std` depends on `cfg-if`, with features: `rustc-dep-of-std` but `cfg-if` does not have these features.


failed to select a version for `cfg-if` which could resolve this conflict
```

But the feature is present since https://github.com/alexcrichton/cfg-if/commit/3e311415b29781b886a6b6611206305cec9527a7.

Reply:
Looks like the crates.io version differs from the GH version in that parts of the toml are commented out...??
```
# [dependencies]
# core = { version = "1.0.0", optional = true, package = 'rustc-std-workspace-core' }
# compiler_builtins = { version = '0.1.2', optional = true }
#
# [features]
# rustc-dep-of-std = ['core', 'compiler_builtins']
```

Reply:
Reported as https://github.com/alexcrichton/cfg-if/issues/22.

Reply:
And submitted hopefully fixing PR has https://github.com/rust-lang/rust/pull/61828.